src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+
+    private final OrderService orderService;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final DtoResponseMapper<OrderResponseDto, Order> orderResponceMapper;
+
+    public OrderController(OrderService orderService,
+                           ShoppingCartService shoppingCartService, UserService userService,
+                           DtoResponseMapper<OrderResponseDto, Order> orderResponceMapper) {
+        this.orderService = orderService;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.orderResponceMapper = orderResponceMapper;
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto completeOrder(@RequestParam Long userId) {
+        Order completedOrder = orderService
+                .completeOrder(shoppingCartService.getByUser(userService.get(userId)));
+        return orderResponceMapper.toDto(completedOrder);
```

Comment:
There is a typo when using the variable name 'orderResponceMapper'. It should be 'orderResponseMapper' 🧐